src/bandersnatch/storage.py

```diff
@@ -0,0 +1,284 @@
+"""
+Storage management
+"""
+import contextlib
+import hashlib
+import logging
+import pathlib
+from collections import defaultdict
+from typing import (
+    IO,
+    Any,
+    Dict,
+    Generator,
+    Iterable,
+    List,
+    Optional,
+    Sequence,
+    Set,
+    Type,
+    Union,
+)
+
+import pkg_resources
+from packaging.utils import canonicalize_name
+
+from .configuration import BandersnatchConfig
+
+PATH_TYPES = Union[pathlib.Path, str]
```

Comment:
https://github.com/python/typing/issues/402 - Seems one day we might have this. Could add comment. No big deal.

Reply:
good suggestion, let me see if this works with the inheritance i set up

Reply:
so `PathLike` doesn't have the `name` attribute among other things, unfortunately :|